### PR TITLE
Docker and OpenAI API Call

### PR DESCRIPTION
**Pull request by manorajesh**
- Added `Dockerfile` for easy development
- Updated `BUILDING.md` to explain the current Docker setup and deployment
- Backend making a POST request to the `gpt3.5-turbo` model for idea generation
- Custom prompts/criteria supported for generation
- API key assumed to be in an environment variable

### CODE REVIEW

File: BUILDING.md [hunk 1]
```diff
@@ -178,3 +178,30 @@ npm start
 Then, open a browser to http://localhost:8080.
 
 [^1]: Oxford English Dictionary
+
+## Running Project Leo in a Docker Container
```
Comment (sahendrickson) [inline BUILDING.md:182]: How about making this as a third option under "## Running Project Leo"?

- Change "There are two ways to run Project Leo:" to "Project Leo can be run in the following ways:"
- Move this section before "[^1] ...", that should stay at the bottom of the page.
- Change the header from ## to ###.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Get the base image from Docker Hub.
+FROM ubuntu:22.04
+
+# Install Apache Maven, Git, Java, Nodejs, and Sass.
+RUN apt update && apt install -y maven git openjdk-17-doc openjdk-17-jdk openjdk-17-source nodejs sass
```
Comment (sahendrickson) [inline Dockerfile:5]: Can you try this without "nodejs" and "sass"? I think "mvn verify" will still work.

**Review by sahendrickson — COMMENTED** [review]
One comment to make your section fit a little better in the documentation and another to see if you can remove some apt install dependencies.

**Comment by manorajesh** [comment]
> One comment to make your section fit a little better in the documentation and another to see if you can remove some apt install dependencies.

Addressed issues

**Comment by manorajesh** [comment]
Should there be some form of logging for `OpenAIController.java`? I kept getting errors when I added a logger which was set up similarly to the one in `ReactResourceController.java`. In the `server/pom.xml`, a comment read, "Log4j only allows one logger implementation to be registered." Can there only be one logger active at once?

**Comment by sahendrickson** [comment]
I'm not sure what's going on with the logger. I may have set it up incorrectly. Let's get what you have in and then we'll figure out those things.

**Review by sahendrickson — APPROVED** [review]

**Pull request merged**